src/main.ts

```diff
@@ -198,7 +202,15 @@ async function drawLoop(): Promise<void> {
     await drawFrame(currentFrame);
     const delta = leftArrowDown ? -1 : 1;
     currentFrame = (currentFrame + delta + dataset.numberOfFrames) % dataset.numberOfFrames;
+    // update higlighted cell id if any
+    if (selectedTrack) {
+      const id = selectedTrack.getIdAtTime(currentFrame);
+      canv.setHighlightedId(id - 1);
```

Comment:
these -1 shifts make me uneasy.  Happens in shader, and when we read back on click.   Would be nice to isolate that somewhere.

Reply:
Could start using the alpha channel as a background flag?

Reply:
Since the shifts only really become necessary when indexing into these big data arrays, I guess part of why we're (necessarily) creating extra confusion around this now is because tracks introduce new places outside the shader where we need to read feature data. I found subtracting 1 immediately when reading back on click to be a bit confusing - it leads to having to remember that the highlighted ID is already shifted back by 1 in the shader:
`if (int(id) - 1 == highlightedId) { ... }`
Both of the shifts you mention could be removed without affecting this comparison. It would require shifts to instead be made in getters for track ids and times, but that might be a reasonable way to, if not isolate shifts, at least standardize where they ought to happen: immediately before using a cell ID to index into a data array. (It's super possible that this would instead introduce more headache, I haven't fully investigated - e.g. it looks like a function like `getIdsOfTrack` would have to start *adding* 1 to its output to shift array indexes back to cell IDs.)
Seeing the -1 cell id used in plot titles also makes me wonder whether scientists would expect cell IDs to start at 1 or 0, or if they'll ever want to use the number currently in the plot title in the first place.

Reply:
Maybe not worth doing anything just yet but if this comes up more often we can do a single PR to resolve it all, referencing the logic you outlined here.